Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -756,6 +755,13 @@ private void ReleaseRigidBody()
             }
         }
 
+        private IMixedRealityPointer GetFirstPointer()
+        {
+            // We may be able to do this without allocating memory.
+            // Moving to a method for later investigation.
+            return pointerIdToPointerMap.Values.First();
```

Comment:
Easiest may literally be to just have two fields: 

```
Pointer firstPointer;
Pointer secondPointer;

OnPointerDown(Pointer newPointer)
{
secondPointer = newPointer
}

OnPointerUp/OnSourceLost(Pointer lostPointer)
{
  if (lostPointer == secondPointer)
  {
     secondPointer = null;
  }
  if (lostPointer == firstPointer)
  {
    firstPointer = secondPointer;
    secondPointer = null;
  }
}
```

This of course assumed you never have more than two pointers, but I think this is fair and we can debug assert that.